core/src/avm1/globals/point.rs

```diff
@@ -0,0 +1,376 @@
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let proto = context.system_prototypes.point;
+    let args = [
+        this.get("x", avm, context)?.resolve(avm, context)?,
+        this.get("y", avm, context)?.resolve(avm, context)?,
+    ];
+    let cloned = proto.new(avm, context, proto, &args)?;
+    let _ = constructor(avm, context, cloned, &args)?;
+
+    Ok(cloned.into())
+}
+
+fn equals<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    if let Some(Value::Object(other)) = args.get(0) {
+        let this_x = this.get("x", avm, context)?.resolve(avm, context)?;
+        let this_y = this.get("y", avm, context)?.resolve(avm, context)?;
+        let other_x = other.get("x", avm, context)?.resolve(avm, context)?;
+        let other_y = other.get("y", avm, context)?.resolve(avm, context)?;
+        return Ok((this_x == other_x && this_y == other_y).into());
+    }
+
+    Ok(false.into())
+}
+
+fn add<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let this_x = this
+        .get("x", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let this_y = this
+        .get("y", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let other = value_to_point(
+        args.get(0).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+    let object = point_to_object((this_x + other.0, this_y + other.1), avm, context)?;
+    Ok(object.into())
+}
+
+fn subtract<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let this_x = this
+        .get("x", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let this_y = this
+        .get("y", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let other = value_to_point(
+        args.get(0).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+    let object = point_to_object((this_x - other.0, this_y - other.1), avm, context)?;
+    Ok(object.into())
+}
+
+fn distance<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    if args.len() < 2 {
+        return Ok(Value::Undefined.into());
+    }
+
+    let a = value_to_point(
+        args.get(0).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+    let b = value_to_point(
+        args.get(1).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
```

Comment:
`distance` doesn't seem to actually work for non-Point objects like `{x: 1, y: 2}` -- uses `subtract`/`length` existing on prototype